closed/src/java.base/share/classes/openj9/internal/security/RestrictedSecurity.java

```diff
@@ -0,0 +1,1053 @@
+                    "-Dsemeru.restrictedsecurity=<n>  This flag will select the settings for the user " +
+                            "specified restricted security policy.");
+            System.out.println(
+                    "-Dsemeru.restrictedsecurity=audit  This flag will list the name and number of all " +
+                            "configured restricted security policies.");
+            System.out.println(
+                    "-Dsemeru.restrictedsecurity=trace  This flag will list all properties relevant to " +
+                            "the restricted security mode, including the existing default properties and the " +
+                            "restricted security properties.");
+            System.out.println("-Dsemeru.restrictedsecurity=help  This flag will print help message.");
+
+            System.out.println();
+            System.out.println("e.g.");
+            System.out.println("    -Dsemeru.restrictedsecurity=1,trace,audit,help");
+            System.out.println("    -Dsemeru.restrictedsecurity=help");
+
+            System.out.println();
+        }
+
+        /**
+         * Check if the input string is null. If null return "".
+         *
+         * @param string the input string
+         * @return "" if the string is null
+         */
+        private static String parseProperty(String string) {
+            return (string != null) ? string.trim() : "";
+        }
+
+        /**
+         * Check if the brackets are balanced.
+         *
+         * @param string input string for checking
+         * @return true if the brackets are balanced
+         */
+        private static boolean areBracketsBalanced(String string) {
+            Deque<Character> deque = new LinkedList<>();
+
+            for (char ch : string.toCharArray()) {
+                switch (ch) {
+                    case '{':
+                        deque.addFirst('}');
+                        break;
+                    case '[':
+                        deque.addFirst(']');
+                        break;
+                    case '(':
+                        deque.addFirst(')');
+                        break;
+                    case '}':
+                    case ']':
+                    case ')':
+                        if (deque.isEmpty() || (deque.removeFirst().charValue() != ch)) {
+                            return false;
+                        }
+                        break;
+                    default:
+                        break;
+                }
+            }
+            return deque.isEmpty();
+        }
+
+        /**
+         * Check if the input string is asterisk (*).
+         *
+         * @param string input string for checking
+         * @return true if the input string is asterisk
+         */
+        private static boolean isAsterisk(String string) {
+            return "*".equals(string);
+        }
+
+        /**
+         * Nested class for provider's constraints
```

Comment:
Please remove "Nested" and expand on the purpose of this class (using full sentences and proper punctuation).